installer/installer.py

```diff
@@ -638,76 +456,41 @@ def cmd_install_docker_cluster_store():
-        store_config(DOCKER_INSTALL_CONFIG, install_config)
-
-        # Update the daemon config.
-        daemon_config["cluster-store"] = CLUSTER_STORE_ETCD_PROXY
-        store_config(DOCKER_DAEMON_CONFIG, daemon_config)
-
-    # If Docker was already configured to use a cluster store, and not by
-    # Calico, exit now.
-    if not install_config.get("docker-updated"):
-        _log.debug("Docker not updated by Calico")
-        return
+    _log.debug("Configure cluster store in daemon config")
+    install_config["docker-updated"] = True
+    store_config(DOCKER_INSTALL_CONFIG, install_config)
 
-    # If Docker config was updated, store the current Docker process creation
-    # time so that we can identify when Docker is restarted.
-    if not install_config.get("docker-created"):
-        _log.debug("Store docker daemon creation time")
-        daemon_process = wait_for_process(DOCKER_DAEMON_EXE_RE,
-                                          DOCKER_DAEMON_PARMS_RE,
-                                          MAX_TIME_FOR_DOCKER_RESTART)
-        install_config["docker-created"] = str(daemon_process.create_time())
-        store_config(DOCKER_INSTALL_CONFIG, install_config)
+    # Set cluster-store in the daemon config.
+    daemon_config["cluster-store"] = cluster_store
+    store_config(DOCKER_DAEMON_CONFIG, daemon_config)
 
     # Check the Docker daemon is running.
-    daemon_process = wait_for_process(DOCKER_DAEMON_EXE_RE,
-                                      DOCKER_DAEMON_PARMS_RE,
-                                      MAX_TIME_FOR_DOCKER_RESTART,
-                                      PROCESS_STABILITY_TIME,
-                                      fail_if_not_found=False)
-    if not daemon_process and \
-       not install_config.get("docker-attempted-start"):
-        # No Docker daemon process - it must have failed to come back.
-        # We haven't tried a systemctl start so try one now.
-        _log.error("Docker process failed after restart - attempt start")
-        start_service("docker")
-        install_config["docker-attempted-restart"] = True
-        store_config(DOCKER_INSTALL_CONFIG, install_config)
-        daemon_process = wait_for_process(DOCKER_DAEMON_EXE_RE,
-                                          DOCKER_DAEMON_PARMS_RE,
-                                          MAX_TIME_FOR_DOCKER_RESTART,
-                                          PROCESS_STABILITY_TIME)
-
-    if install_config["docker-created"] == str(daemon_process.create_time()):
-        # Docker has not been restarted, so restart it now.  This will cause
-        # the task to fail (but sys.exit(1) to make sure).  The task will be
-        # relaunched, and next time will miss this branch and succeed.
-        _log.warning("Restarting Docker process: %s", daemon_process)
-        restart_service("docker", daemon_process)
-        sys.exit(1)
+    _log.info("Restarting Docker...")
+    restart_service(DOCKER_SERVICE_NAME)
```

Comment:
We should store in the `install_config` when we have restarted the service.  Is there not a possibility that restarting docker could cause the agent to fail which means you'd get into a cycle of restarting.

Perhaps not with a standard install of DCOS - but maybe with a containerized version.